testing/chain.go

```diff
@@ -60,8 +59,15 @@ type TestChain struct {
 	TxConfig      client.TxConfig
 	Codec         codec.BinaryCodec
 
-	Vals    *tmtypes.ValidatorSet
-	Signers []tmtypes.PrivValidator
+	Vals     *tmtypes.ValidatorSet
+	NextVals *tmtypes.ValidatorSet
+
+	// Signers is a map from validator address to the PrivValidator
+	// The map is converted into an array that is the same order as the validators right before signing commit
```

Comment:
Where do you ensure the ordering is the same? Go maps don't use any sorted ordering. You'd need to sort the array

Reply:
We convert the unordered map into ordered array here: https://github.com/cosmos/ibc-go/blob/aditya%2Fupdate-val-set/testing/chain.go#L467